tests/test_storm_control.py

```diff
@@ -0,0 +1,528 @@
+        time.sleep(1)
+
+    def add_port_channel_member(self, dvs, lag_name, member):
+        dvs.runcmd("config portchannel member add "+ lag_name + " "+ member)
+        time.sleep(1)
+
+    def create_vlan(self, dvs, vlan):
+        dvs.runcmd("config vlan add " + vlan)
+        time.sleep(1)
+
+    def add_vlan_member(self, dvs, vlan, interface):
+        dvs.runcmd("config vlan member add " + vlan + " " + interface)
+        time.sleep(1)
+
+    def add_storm_session(self, if_name, storm_type, kbps_value):
+        tbl = swsscommon.Table(self.cdb, "PORT_STORM_CONTROL")
+        fvs = swsscommon.FieldValuePairs([("kbps", str(kbps_value))])
+        key = if_name + "|" + storm_type
+        tbl.set(key,fvs)
+        time.sleep(1)
+
+    def delete_storm_session(self, if_name, storm_type):
+        tbl = swsscommon.Table(self.cdb, "PORT_STORM_CONTROL")
+        key = if_name + "|" + storm_type
+        tbl._del(key)
+        time.sleep(1)
+
+    def test_add_bcast_storm(self,dvs,testlog):
+        self.setup_db(dvs)
+
+        if_name = "Ethernet0"
+        storm_type = "broadcast"
+        #User input is Kbps
+        #Orchagent converts the value to CIR as below and programs the ASIC DB
+        #kbps_value * 1000 / 8
+        kbps_value = 1000000
+
+        self.add_storm_session(if_name, storm_type, kbps_value)
+
+        tbl = swsscommon.Table(self.cdb,"PORT_STORM_CONTROL")
+        (status,fvs) = tbl.get(if_name+"|"+storm_type)
+
+        assert status == True
+        assert len(fvs) > 0
+
+        port_oid = dvs.asicdb.portnamemap[if_name]
+
+        atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_PORT")
+        status, fvs = atbl.get(dvs.asicdb.portnamemap[if_name])
+        assert status == True
+        print ("\nPort field-value pairs")
+        print (fvs)
+
+        policer_oid = 0
+
+        for fv in fvs:
+            if fv[0] == "SAI_PORT_ATTR_BROADCAST_STORM_CONTROL_POLICER_ID":
+                policer_oid = fv[1]
+
+        if policer_oid != 0:
+            atbl = swsscommon.Table(self.adb,"ASIC_STATE:SAI_OBJECT_TYPE_POLICER")
+            status, fvs = atbl.get(policer_oid)
+            assert status == True
+            print ("POlicer parameters")
+            print (fvs)
+
+        bps = 0
+
+        for fv in fvs:
+            if fv[0] == "SAI_POLICER_ATTR_CIR":
+                bps = fv[1]
+
+        #Retrieved value of bps from ASIC_DB is converted back to user input kbps
+        kbps = int(bps) / int(1000) * 8
+        print ("KBPS value")
+        print (kbps)
+
+        assert str(kbps) == str(kbps_value)
+
+    def test_add_uucast_storm(self,dvs,testlog):
+        self.setup_db(dvs)
+
+        if_name = "Ethernet0"
+        storm_type = "unknown-unicast"
+        #User input is Kbps
+        #Orchagent converts the value to CIR as below and programs the ASIC DB
+        #kbps_value * 1000 / 8
+        kbps_value = 1000000
+
+        self.add_storm_session(if_name, storm_type, kbps_value)
```

Comment:
replace all the code in this method with call to add_storm_control_interface()

Reply:
Done.